### ISSUE TITLE
getCarrierName crash

### ISSUE THREAD

**Issue by AlexAndriichenko** [OneSignal/OneSignal-Android-SDK#427]
I have crashes in 3.7.1 version. READ_PHONE_STATE permission access is not handled
```java
Fatal Exception: java.lang.SecurityException: Requires READ_PHONE_STATE: Neither user 10390 nor current process has android.permission.READ_PHONE_STATE.
       at android.os.Parcel.readException(Parcel.java:1546)
       at android.os.Parcel.readException(Parcel.java:1499)
       at com.android.internal.telephony.IPhoneSubInfo$Stub$Proxy.getSubscriberIdForSubscriber(IPhoneSubInfo.java:756)
       at android.telephony.TelephonyManager.getSubscriberId(TelephonyManager.java:2009)
       at android.telephony.TelephonyManager.getNetworkOperatorName(TelephonyManager.java:1122)
       at android.telephony.TelephonyManager.getNetworkOperatorName(TelephonyManager.java:1094)
       at com.onesignal.OSUtils.getCarrierName(OSUtils.java:140)
       at com.onesignal.OneSignal$7.run(OneSignal.java:1035)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by jkasten2** [comment]
@AlexAndriichenko We added a catch for this and it will be included in the next release. Could you let us know which device models and Android versions you seen this crash on?

**Comment by disav1** [comment]
Zenfone Go (ASUS_X005)100%
Operating Systems>100% 5
5.1

**Comment by jkasten2** [comment]
@disav1 Thanks for the details, this has been completely fixed in the 3.8.0 update

**Comment by disav1** [comment]
Do you add READ_PHONE_STATE to manifest? Or just catch the exception?
